action/protocol/staking/handlers.go

```diff
@@ -313,6 +313,11 @@ func (p *Protocol) handleChangeCandidate(ctx context.Context, act *action.Change
 		}
 	}
 
+	if p.hu.IsPost(config.Hawaii, blkCtx.BlockHeight) && address.Equal(prevCandidate.Owner, candidate.Owner) {
```

Comment:
we can do this earlier by comparing bucket.Candidate and candidate.Owner, before loading "prevCandidate".

Reply:
line 309 above: if bucket is unstaked and after Greenland, it will return a diff error

this check needs to be after that check, otherwise block won't commit in case the tx meets all these conditions

Reply:
actually block can commit, but it's inconsistent behavior for such block before Hawaii (return error in Greenland fix) vs. after Hawaii (return nil)